internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -1058,3 +1072,86 @@ func (a *Adapter) findSnapshotWithOpenedPR(snapshot *applicationapiv1alpha1.Snap
 	}
 	return nil, nil
 }
+
+// checkOldSnapshotsPipelineRun sorts all snapshots for application and cancells all running interationTest pipelineruns within application
+// removes finalizer before the pipelinerun is set as cancelled run finally to be gracefuly cancelled
+// god please forgive me for this monstrosity of cognitive complexity 40+
```

Comment:
Hopefully it should be now way lower than 40 after adjustments :laughing:

Reply:
Let me remove this, its 39 now :(.